library/src/main/java/com/google/android/exoplayer/extractor/webm/WebmExtractor.java

```diff
@@ -235,6 +241,7 @@
   private int sampleBytesWritten;
   private boolean sampleRead;
   private boolean sampleSeenReferenceBlock;
+  private Inflater sampleInflator;
```

Comment:
Content compression is a per-track setting. The test file i'm using myself contains a H264 and DTS track (uncompressed) and two separate zlib compressed PGS tracks. I generated my own test file using mkvtoolnix (https://mkvtoolnix.download/) which allows the compression to be specified.

I assume you're asking this question to query why I made the `sampleInflator` scoped for the extractor, and not one per track (inside the `Track` instance)? I guess my assumption was that each individual sample is read independently, and we don't need to worry about asynchronously reading two samples at the same time (is this valid?). Since `resetSample` is called between sample reads from different tracks, the inflator instance is reset.

The alternative would be to create an Inflator instance per-compressed-track and keep it around. We would then re-use it for all sample reads rather than using the one inside the Extractor which keeps on being reset/recreated between samples of different tracks.

Is that preferable? I'm happy to change this as you see best, just want to make sure I understand before making (and testing) the changes.